pkg/build/api/v1/types.go

```diff
@@ -371,6 +375,13 @@ type SourceBuildStrategy struct {
 	ForcePull bool `json:"forcePull,omitempty" description:"forces the source build to pull the image if true"`
 }
 
+// BuildPostCommitSpec holds a build post commit hook specification.
+type BuildPostCommitSpec struct {
+	// RunBash holds a bash script to be executed in a container running the
+	// build output image.
+	RunBash string `json:"runBash,omitempty" description:"bash script to be executed in a container running the build output image"`
```

Comment:
> Can we apply this to our example patterns and verify this will work for the base cases?

Yes.


> args: ["bundle", "exec", "rake", "test"] runs /bin/bash -c "bundle exec rake test"

Shouldn't this be `args: ["bundle exec rake test"]`? (`/bash -c` takes 1 argument)

> command: ["/foo"] - runs /foo in the container, no shell

What does `command: ["ls"]` do? Does it run in a shell?

--------------

Generally, is this proposed solution a pattern taken from somewhere I can refer to, or is it something we're coming up with?

Are we trying to strictly mimic Docker's CMD and ENTRYPOINT instructions in their JSON array forms?
https://docs.docker.com/engine/reference/builder/#cmd
https://docs.docker.com/engine/reference/builder/#entrypoint

In Docker, if CMD is `["bundle", "exec", "rake", "test"]`, like `args` in our example above, and ENTRYPOINT is not `["/some/shell", "-c"]`, the command will not run in a shell.